tests/PHPStan/Rules/Functions/CallToFunctionParametersRuleTest.php

```diff
@@ -554,4 +554,66 @@ public function testBugNumberFormatNamedArguments(): void
 		$this->analyse([__DIR__ . '/data/number-format-named-arguments.php'], []);
 	}
 
+	public function testArrayReduceCallback(): void
+	{
+		$this->analyse([__DIR__ . '/data/array_reduce.php'], [
+			[
+				'Parameter #2 $callback of function array_reduce expects callable(string, int|string): string, Closure(string, string): string given.',
+				5,
+			],
+			[
+				'Parameter #2 $callback of function array_reduce expects callable(string|null, int): string|null, Closure(string, int): string given.',
+				13,
+			],
+			[
+				'Parameter #2 $callback of function array_reduce expects callable(string|null, int): string|null, Closure(string, int): string given.',
```

Comment:
This is the last remaining test failure. If no argument is passed, then the first function call will take null, as the `carry` param.

Not sure if we should ignore this issue for now, or drop array_reduce as well

Reply:
This highlights a separate issue that's not related to generics. The problem is that we don't know the default value of parameters for built-in functions. For `array_reduce` this place should pass in `new NullType()` instead of `null`: https://github.com/phpstan/phpstan-src/blob/2abb92e2588808b3a8dd5cc3206041b7afa9891d/src/Reflection/SignatureMap/NativeFunctionReflectionProvider.php#L103 (you can try to edit it locally and see if the test passes).

I don't think we have to block this PR on this so feel free to comment out the test and I'll think of something. We might be able to extract this information from phpstorm-stubs and https://github.com/phpstan/php-8-stubs, but that's a completely separate feature request...

Reply:
I've fixed this in a similar matter as fputcsv, and sprintf.